### PR TITLE
DTSPO-440 Add Webhook

### PR DESCRIPTION
**Pull request by zc-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: terraform/runbook.tf [hunk 1]
```diff
@@ -12,7 +12,7 @@ resource "azurerm_automation_account" "github_membership_automation" {
 }
 
 resource "azurerm_automation_runbook" "github_membership_runbook" {
-  name                    = "github_membership_runbook"
+  name                    = "${var.runbook_name}_runbook"
```
Comment (slaterin) [inline terraform/runbook.tf:15]: Do we need runbook in the name?

Comment (msl8r) [inline terraform/runbook.tf:15]: Do we need runbook in the name?

File: terraform/webhook-template.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "resources": [
+        {
+        "name": "${azurerm_automation_account.github_membership_automation.name}/webhook",
+        "type": "Microsoft.Automation/automationAccounts/webhooks",
+        "apiVersion": "2015-10-31",
+        "properties": {
+            "isEnabled": true,
+            "uri": "https://${var.runbook_name}.webhook.uks.azure-automation.net/webhooks?token=${random_string.token1.result}%2b${random_string.token2.result}%3d",
```
Comment (msl8r) [inline terraform/webhook-template.tf:30]: Let's make the whole uri a local value and pass a single variable

File: terraform/webhook-template.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
+resource "azurerm_template_deployment" "github_management_template" {
```
Comment (timja) [inline terraform/webhook-template.tf:1]: shouldn't you use a this:
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group_template_deployment

this one is deprecated

**Review by msl8r — APPROVED** [review]
LGTM

File: terraform/webhook-template.tf (hunk 3 above)
Comment (zc-hmcts) [reply terraform/webhook-template.tf]: ammended to use this instead

**Pull request merged**